actor/src/macros.rs

```diff
@@ -0,0 +1,122 @@
+/// Constructs an envelope enumeration that contains all messages for an actor.
+///
+/// The first identifier is the name of the enum.
+/// The second identifier is the name of a trait specific to the actor.
+/// The remaining pairs are the variants of the envelope indicating the messages the actor handles.
+///
+/// The constructed trait is a union of the [`crate::Handler`] traits for each message along with the [`crate::Actor`] trait.
+#[macro_export]
+macro_rules! actor_envelope {
+    (
+        $enum_name:ident,
+        $trait_name:ident,
+        $(
+            $variant_name:ident => $message_type:ty,
+        )*
+    ) => {
+        /// Wrapping envelope for all messages of the actor.
+        pub enum $enum_name {
+            $(
+                #[doc = stringify!($variant_name)]
```

Comment:
rust analyzer seems to have a hard time displaying these `#[doc]` macro comments. I hoped something like this would make a difference but no luck

```rust
#[doc = concat!(
                    stringify!($variant_name),
                    " contains messages of type [`",
                    stringify!($message_type),
                    "`]."
                )]
```

Reply:
yeah I saw this too, I found this https://users.rust-lang.org/t/trying-to-generate-doc-comments-with-macros/81783, but it doesn't seem to work. And it doesn't complain either.

Reply:
I haven't been able to figure this out. The generated docs are not that important. Maybe this a regression in recent versions of rust? Either way I am going to leave it in place for now.